oss-internship-2020/libtiff/test/defer_strile_writing.cc

```diff
@@ -0,0 +1,327 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+
+#include "helper.h"
+#include "tiffio.h"  // NOLINT(build/include)
+
+// sapi functions:
+//  TIFFOpen
+//  TIFFClose
+//  TIFFGetField
+//  TIFFSetField
+//  TIFFWriteCheck
+//  TIFFSetDirectory
+//  TIFFFreeDirectory
+//  TIFFWriteScanline
+//  TIFFWriteDirectory
+//  TIFFCreateDirectory
+//  TIFFReadEncodedTile
+//  TIFFReadEncodedStrip
+//  TIFFWriteEncodedTile
+//  TIFFWriteEncodedStrip
+//  TIFFDeferStrileArrayWriting
+//  TIFFForceStrileArrayWriting
+
+namespace {
+
+#define TBS 256  // kTileBufferSize
```

Comment:
Why is this a define?

Reply:
like SPP, it is used for std::array<unsigned char, TBS>

Reply:
Can we use `constexpr size_t kTileBufferSize = 256` instead? It seems you already added that as a comment?